.github/workflows/pypi_publish.yml

```diff
@@ -16,32 +19,101 @@ jobs:
       - uses: hynek/build-and-inspect-python-package@v2
 
   publish:
-    name: Publish to PyPI
+    name: Publish CDK to PyPI
     runs-on: ubuntu-latest
     needs: [build]
     permissions:
       id-token: write # IMPORTANT: this permission is mandatory for trusted publishing
       contents: write # Needed to upload artifacts to the release
-    environment:
-      name: PyPI
-      url: "https://pypi.org/p/airbyte-cdk"
-    if: startsWith(github.ref, 'refs/tags/v')
+    if: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'workflow_dispatch'
+
     steps:
-      - uses: actions/download-artifact@v4
-        with:
-          name: Packages
-          path: dist
+      - name: Debug Reference
+        run: |
+          echo "Event Name: $GITHUB_EVENT_NAME"
+          echo "GitHub Ref: $GITHUB_REF"
+
+      - name: Validate Tag or Version Input
+        run: |
+          if [[ "$GITHUB_EVENT_NAME" == "workflow_dispatch" ]]; then
+            if [ -z "${{ github.event.inputs.version }}" ]; then
+              echo "Error: No version input provided for workflow_dispatch."
+              exit 1
+            fi
+            echo "VERSION=${{ github.event.inputs.version }}" >> $GITHUB_ENV
+          elif [[ "$GITHUB_REF" =~ ^refs/tags/v(.+)$ ]]; then
+            VERSION=${BASH_REMATCH[1]}
+            echo "VERSION=$VERSION" >> $GITHUB_ENV
+          else
+            echo "Error: This workflow requires a valid tag (refs/tags/vX.Y.Z) to run."
+            exit 1
+          fi
+          echo "Validated Version: $VERSION"
+
       - name: Attach Wheel to GitHub Release
         uses: svenstaro/upload-release-action@v2
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
           file: dist/*.whl
-          tag: ${{ github.ref }}
+          tag: ${{ env.VERSION }}
           overwrite: true
           file_glob: true
```

Comment:
@ChristoGrab - This step will only work with release tag events.

You can add an "if" condition like `if: startsWith(github.ref, 'refs/tags/v')` to skip it for workflow dispatch.

The reason is that it's job is to attach the wheel to the release (not strictly necessary anyway). If using the `VERSION` env var, the "tag" format will probably need to be reconstructed: `refs/tags/v${VERSION}`.